webapp/src/components/AcceptBidAssetPage/AcceptBidAssetPage.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import { ASSET_TYPES } from 'shared/asset'
+import AcceptBidParcelPage from './AcceptBidParcelPage'
+import AcceptBidEstatePage from './AcceptBidEstatePage'
+import { bidType } from 'components/types'
+
+export default class AcceptBidAssetPage extends React.PureComponent {
+  static propTypes = {
+    id: PropTypes.string.isRequired,
+    bid: bidType,
+    assetType: PropTypes.string.isRequired,
+    isConnected: PropTypes.bool,
+    isLoading: PropTypes.bool.isRequired
+  }
+
+  bidFetched = false
+
+  componentWillReceiveProps(nextProps) {
+    const { isConnected, isLoading, onFetchBidByAsset, bid } = nextProps
+
+    if (!isConnected || isLoading || bid || this.bidFetched) {
+      return
+    }
+    this.bidFetched = true
+
+    onFetchBidByAsset()
+  }
+
+  render() {
+    const { assetType, bid } = this.props
+    if (!bid) {
+      return null
+    }
+
+    switch (assetType) {
+      case ASSET_TYPES.parcel:
+        return <AcceptBidParcelPage {...this.props} />
+      case ASSET_TYPES.estate:
+        return <AcceptBidEstatePage {...this.props} />
+      default:
+        return null
+    }
+  }
+}
```

Comment:
I noticed that the render (not renderPage) and handleConfirm methods of each `AcceptBid...Page` are the same. Maybe they belong on this file?